crates/oxc_semantic/tests/integration/const_enum.rs

```diff
@@ -0,0 +1,391 @@
+    assert_const_enum_value(&multiply_member.value, "12");
+
+    // Check Divide member (should be "5")
+    let divide_member = find_member_by_name(enum_info, "Divide").unwrap();
+    assert_const_enum_value(&divide_member.value, "5");
+
+    // Check Negate member (should be "-5")
+    let negate_member = find_member_by_name(enum_info, "Negate").unwrap();
+    assert_const_enum_value(&negate_member.value, "-5");
+
+    // Check Plus member (should be "7")
+    let plus_member = find_member_by_name(enum_info, "Plus").unwrap();
+    assert_const_enum_value(&plus_member.value, "7");
+
+    // Check Not member (should be "false")
+    let not_member = find_member_by_name(enum_info, "Not").unwrap();
+    assert_const_enum_value(&not_member.value, "false");
+
+    // Check Shift member (should be "4", 1 << 2)
+    let shift_member = find_member_by_name(enum_info, "Shift").unwrap();
+    assert_const_enum_value(&shift_member.value, "4");
+
+    // Check Bitwise member (should be "7", 5 | 3 = 101 | 011 = 111)
+    let bitwise_member = find_member_by_name(enum_info, "Bitwise").unwrap();
+    assert_const_enum_value(&bitwise_member.value, "7");
+}
+
+#[test]
+fn test_const_enum_constant_propagation() {
+    let source = "
+            const enum Values {
+                A = 1,
+                B = A,
+                C = A + 2,
+                D = B * 3,
+                E = C + D,
+                F = A + B + C + D
+            }
+        ";
+    let allocator = Allocator::default();
+    let source_type: SourceType = SourceType::default().with_typescript(true);
+    let semantic = get_semantic(&allocator, source, source_type);
+
+    // Find the Values enum
+    let values_enum = semantic
+        .const_enums()
+        .enums()
+        .find(|(_, enum_info)| semantic.scoping().symbol_name(enum_info.symbol_id) == "Values");
+
+    assert!(values_enum.is_some());
+
+    let (_, enum_info) = values_enum.unwrap();
+
+    // Check enum members
+    assert_eq!(enum_info.members.len(), 6);
+
+    // A should be "1"
+    let a_member = find_member_by_name(enum_info, "A").unwrap();
+    assert_const_enum_value(&a_member.value, "1");
+
+    // B should be "1" (references A - constant propagation works for simple references)
+    let b_member = find_member_by_name(enum_info, "B").unwrap();
+    assert_const_enum_value(&b_member.value, "1");
+
+    // C should be "3" (A + 2 = 1 + 2 - constant propagation works in expressions)
+    let c_member = find_member_by_name(enum_info, "C").unwrap();
+    assert_const_enum_value(&c_member.value, "3");
+
+    // D should be "3" (B * 3 = 1 * 3 - constant propagation works in expressions)
+    let d_member = find_member_by_name(enum_info, "D").unwrap();
+    assert_const_enum_value(&d_member.value, "3");
+
+    // E: Currently "Computed" because C and D are computed values
+    // TODO: Should be "6" (C + D = 3 + 3) when full constant propagation is implemented
```

Comment:
This is because the current https://github.com/oxc-project/oxc/blob/d8881ae8ad32570052955d31198ecbb73a6c5bf8/crates/oxc_ecmascript/src/constant_evaluation/mod.rs?plain=1#L212-L236 does not support the scenario where an identifier reference points to a const value.